platform/linuxbsd/os_linuxbsd.cpp

```diff
@@ -212,39 +212,40 @@ String OS_LinuxBSD::get_name() const {
 }
 
 String OS_LinuxBSD::get_systemd_os_release_info_value(const String &key) const {
-	static String info;
-	if (info.is_empty()) {
-		Ref<FileAccess> f = FileAccess::open("/etc/os-release", FileAccess::READ);
-		if (f.is_valid()) {
-			while (!f->eof_reached()) {
-				const String line = f->get_line();
-				if (line.find(key) != -1) {
-					return line.split("=")[1].strip_edges();
-				}
+	Ref<FileAccess> f = FileAccess::open("/etc/os-release", FileAccess::READ);
+	if (f.is_valid()) {
+		while (!f->eof_reached()) {
+			const String line = f->get_line();
+			if (line.find(key) != -1) {
+				String value = line.split("=")[1].strip_edges();
+				value = value.trim_prefix("\"");
+				return value.trim_suffix("\"");
 			}
 		}
 	}
-	return info;
+	return "";
 }
 
 String OS_LinuxBSD::get_distribution_name() const {
-	static String systemd_name = get_systemd_os_release_info_value("NAME"); // returns a value for systemd users, otherwise an empty string.
-	if (!systemd_name.is_empty()) {
-		return systemd_name;
+	static String distribution_name = get_systemd_os_release_info_value("NAME"); // returns a value for systemd users, otherwise an empty string.
+	if (!distribution_name.is_empty()) {
+		return distribution_name;
 	}
```

Comment:
`static` is used for something which is global/reusable in the class, as a cache system. So if you always re-assign it a new string, there's no point in making it static. It's only useful in this kind of construct:
```cpp
static String some_string;
if (some_string.is_empty()) { // True the first time, won't be true next time so we skip the calculation.
    some_string = some_complex_method();
}
return some_string;
```

Here `static` can make sense to avoid opening OS files repeatedly for information that won't change, but it's a bit trickier as you also have a fallback system for non-systemd systems. So the whole logic should be something like this:

```cpp
static String distribution_name;
if (!distribution_name.is_empty()) {
	return distribution_name;
}

// Check first if systemd has the info.
distribution_name = get_systemd_os_release_info_value("NAME");
if (!distribution_name.is_empty()) {
	return distribution_name;
}

// Otherwise utsname can provide a decent value for BSD family.
struct utsname uts; 
uname(&uts);
distribution_name = uts.sysname;
if (!distribution_name.is_empty()) {
	return distribution_name;
}

// Just in case all failed, assign a fallback.
distribution_name = "<unknown>";

return distribution_name;
```

Same for `get_version`.

Reply:
here `get_systemd_os_release_info_value("NAME");` will only be called once, because it's part of the declaration (tested this behaviour, just in case).

but I can adjust to your proposal, which is easier to digest

Reply:
Oh I didn't know it would behave this way. I'm fine either way then.